tests/std/tests/P0083R3_splicing_maps_and_sets/test.cpp

```diff
@@ -634,6 +658,143 @@ void test_unordered_set() {
     test_merge<Set, std::unordered_multiset>();
 }
 
+void test_gh1309() {
+    // Guard against regression of GH-1309, in which node handles were incorrectly destroying the user value with a node
+    // allocator rather than a value_type allocator as the Standard requires.
+
+    {
+        using A  = tracked_allocator<std::pair<int const, char>>;
+        using M  = std::map<int, char, std::less<>, A>;
+        using NH = decltype(std::declval<M&>().extract(42));
+        allocation_guard guard{true};
+        NH nh1;
+        NH nh2;
+        {
+            M m{{0, 'x'}, {1, 'y'}};
+            nh1 = m.extract(0);
+            nh2 = m.extract(1);
+        }
+        construct_destroy_exact = true;
+        nh1                     = std::move(nh2);
+    }
+    construct_destroy_exact = false;
+
+    {
+        using A  = tracked_allocator<std::pair<int const, char>>;
+        using M  = std::multimap<int, char, std::less<>, A>;
+        using NH = decltype(std::declval<M&>().extract(42));
+        allocation_guard guard{true};
+        NH nh1;
+        NH nh2;
+        {
+            M m{{0, 'x'}, {0, 'y'}};
+            nh1 = m.extract(0);
+            nh2 = m.extract(0);
+        }
+        construct_destroy_exact = true;
+        nh1                     = std::move(nh2);
+    }
+    construct_destroy_exact = false;
+
+    {
+        using S  = std::set<int, std::less<>, tracked_allocator<int>>;
+        using NH = decltype(std::declval<S&>().extract(42));
+        allocation_guard guard{true};
+        NH nh1;
+        NH nh2;
+        {
+            S s{{0, 1}};

```

Comment:
Should we use a single layer of braces to construct `set`? The extra layer of braces appears to be inconsistent with `map` construction above. Occurs below.